ecl/hthor/hthor.cpp

```diff
@@ -4239,6 +4239,81 @@ const void * CHThorSortedActivity::nextGE(const void * seek, unsigned numFields)
 
 //=====================================================================================================
 
+CHThorTraceActivity::CHThorTraceActivity(IAgentContext &_agent, unsigned _activityId, unsigned _subgraphId, IHThorTraceArg &_arg, ThorActivityKind _kind)
+: CHThorSteppableActivityBase(_agent, _activityId, _subgraphId, _arg, _kind),
+  helper(_arg),  keepLimit(0), skip(0), sample(0), traceEnabled(false)
+{
+}
+
+void CHThorTraceActivity::ready()
+{
+    CHThorSimpleActivityBase::ready();
+    traceEnabled = agent.queryWorkUnit()->getDebugValueBool("traceEnabled", false);
+    keepLimit = helper.getKeepLimit();
+    if (keepLimit==(unsigned) -1)
+        keepLimit = agent.queryWorkUnit()->getDebugValueInt("traceLimit", 10);
+    if (!traceEnabled)
+        keepLimit = 0;
+    if (!helper.canMatchAny())
```

Comment:
This test should be before the one above, otherwise I don't think it has an effect.